nav2_regulated_pure_pursuit_controller/src/collision_checker.cpp

```diff
@@ -86,10 +86,12 @@ bool CollisionChecker::isCollisionImminent(
   }
 
   const geometry_msgs::msg::Point & robot_xy = robot_pose.pose.position;
-  geometry_msgs::msg::Pose2D curr_pose;
-  curr_pose.x = robot_pose.pose.position.x;
-  curr_pose.y = robot_pose.pose.position.y;
-  curr_pose.theta = tf2::getYaw(robot_pose.pose.orientation);
+  geometry_msgs::msg::Pose curr_pose;
+  curr_pose.position.x = robot_pose.pose.position.x;
+  curr_pose.position.y = robot_pose.pose.position.y;
+  tf2::Quaternion q;
+  q.setRPY(0, 0, tf2::getYaw(robot_pose.pose.orientation));
+  curr_pose.orientation = tf2::toMsg(q);
```

Comment:
Questions:

1. Are there other places we should also strip out the Roll and Pitch values from and/or Poses obtained from TF that we may want to make sure we bring into 2D?
2. If so, should we make a nav2_util for this to remove boilerplate?